tycho-versions-plugin/src/main/java/org/eclipse/tycho/versions/engine/ProjectMetadataReader.java

```diff
@@ -48,49 +48,41 @@ public void reset() {
         projects.clear();
     }
 
-    public PomFile addBasedir(File basedir, boolean recursive) throws IOException {
+    public PomFile addBasedir(File file, boolean recursive) throws IOException {
         // Unfold configuration inheritance
-
-        if (!basedir.exists()) {
-            log.info("Project does not exist at " + basedir);
+        if (!file.exists()) {
+            log.info("Project does not exist at " + file);
             return null;
         }
-        List<ModelProcessor> modelprocessors;
-        try {
-            modelprocessors = container.lookupList(ModelProcessor.class);
-        } catch (ComponentLookupException e) {
-            throw new IOException("can't lookup ModelProcessors");
-        }
-        // normalize basedir to allow modules that explicitly point at pom.xml file
 
-        if (basedir.isFile()) {
-            basedir = basedir.getParentFile();
+        File pomFile;
+        File baseDir;
+        if (file.isFile()) {
+            pomFile = file;
+            baseDir = file.getParentFile();
+        } else {
+            pomFile = lookupPomFile(file);
+            baseDir = file;
         }
 
-        if (projects.containsKey(basedir)) {
+        if (projects.containsKey(baseDir)) {
             return null;
         }
 
-        ProjectMetadata project = new ProjectMetadata(basedir);
-        File pomFile = null;
-        for (ModelProcessor modelProcessor : modelprocessors) {
-            File locatePom = modelProcessor.locatePom(basedir);
-            if (basedir.exists()) {
-                pomFile = locatePom;
-                break;
-            }
-        }
-        if (pomFile == null || !pomFile.exists() || pomFile.length() == 0) {
-            log.warn("No pom file found at " + basedir);
+        if (isValidPomFile(pomFile)) {
```

Comment:
Should this not be
```suggestion
        if (!isValidPomFile(pomFile)) {
```
?

Reply:
No, the condition is correct. But the method name is wrong. it should be "isInvalidPomFile(...)"